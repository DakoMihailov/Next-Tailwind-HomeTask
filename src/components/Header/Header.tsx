/* eslint-disable jsx-a11y/aria-role */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react/react-in-jsx-scope */
import { NextPage } from 'next'

interface Prop {
  setToastStatus: () => void
}
const Header: NextPage<Prop> = ({ setToastStatus }) => (
  <div className='fixed flex flex-row justify-between items-center border-top border-t-2 border-secondary h-rd shadow ease-out duration-300 w-full'>
    <div className='w-full flex justify-center'>
      <div className='flex justify-between items-center gap-4'>
        <svg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'>
          <path d='M5 13L9 17L19 7' stroke='#33A6BA' strokeWidth='2' strokeLinecap='round' strokeLinejoin='round' />
        </svg>
        <div className='font-normal font-5xl font-info' role='maintext'>
          The Location has been updated
        </div>
      </div>
    </div>
    <div className='mr-8 cursor-pointer' onClick={setToastStatus}>
      <svg width='15' height='14' viewBox='0 0 15 14' fill='none' xmlns='http://www.w3.org/2000/svg'>
        <path d='M1.16406 13L13.4928 1L1.16406 13ZM1.16406 1L13.4928 13L1.16406 1Z' fill='#989EA7' />
        <path
          d='M1.16406 1L13.4928 13M1.16406 13L13.4928 1L1.16406 13Z'
          stroke='#989EA7'
          strokeWidth='2'
          strokeLinecap='round'
          strokeLinejoin='round'
        />
      </svg>
    </div>
  </div>
)

export default Header
