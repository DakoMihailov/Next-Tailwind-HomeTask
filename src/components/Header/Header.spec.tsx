import React from 'react'
import { render, screen } from '@testing-library/react'

import Header from './Header'

// Just a sample test.
describe('Header Component', () => {
  it('renders correctly', () => {
    const { container } = render(<Header setToastStatus={() => {}} />)
    expect(container).toMatchSnapshot()
  })

  it('renders The Location has been updated', () => {
    // Given
    const text = 'The Location has been updated'
    render(<Header setToastStatus={() => {}} />)
    // When
    const actual = screen.getByRole('maintext')

    // Then
    expect(actual).toBeInTheDocument()
    expect(actual.textContent).toBe(text)
    expect(actual.className).toBe('font-normal font-5xl font-info')
  })
})
