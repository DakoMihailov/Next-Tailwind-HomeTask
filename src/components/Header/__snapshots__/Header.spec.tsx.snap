// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<div>
  <div
    class="fixed flex flex-row justify-between items-center border-top border-t-2 border-secondary h-rd shadow ease-out duration-300 w-full"
  >
    <div
      class="w-full flex justify-center"
    >
      <div
        class="flex justify-between items-center gap-4"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 13L9 17L19 7"
            stroke="#33A6BA"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <div
          class="font-normal font-5xl font-info"
          role="maintext"
        >
          The Location has been updated
        </div>
      </div>
    </div>
    <div
      class="mr-8 cursor-pointer"
    >
      <svg
        fill="none"
        height="14"
        viewBox="0 0 15 14"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.16406 13L13.4928 1L1.16406 13ZM1.16406 1L13.4928 13L1.16406 1Z"
          fill="#989EA7"
        />
        <path
          d="M1.16406 1L13.4928 13M1.16406 13L13.4928 1L1.16406 13Z"
          stroke="#989EA7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Index Page renders correctly 1`] = `
<div>
  <div
    class="fixed flex flex-row justify-between items-center border-top border-t-2 border-secondary h-rd shadow ease-out duration-300 w-full"
  >
    <div
      class="w-full flex justify-center"
    >
      <div
        class="flex justify-between items-center gap-4"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 13L9 17L19 7"
            stroke="#33A6BA"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <div
          class="font-normal font-5xl font-info"
        >
          The Location has been updated
        </div>
      </div>
    </div>
    <div
      class="mr-8 cursor-pointer"
    >
      <svg
        fill="none"
        height="14"
        viewBox="0 0 15 14"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.16406 13L13.4928 1L1.16406 13ZM1.16406 1L13.4928 13L1.16406 1Z"
          fill="#989EA7"
        />
        <path
          d="M1.16406 1L13.4928 13M1.16406 13L13.4928 1L1.16406 13Z"
          stroke="#989EA7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</div>
`;
